### PR TITLE
#BN-17 feat: create a search endpoint

### PR DESCRIPTION
**Pull request by Niyobuzima**
# PR Description
Please include a brief description of the pull request you want to raise.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
-added a search route
-added product control
# Description of tasks that were expected to be completed and that are not yet completed

# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`

# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

**Review by kunda4 — DISMISSED** [review]

File: .github/workflow/CI.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run linter
+        run: npm run lint
+
+      - name: Run tests
+        run: npm test
```
Comment (dushimeemma) [inline .github/workflow/CI.yml]: add a line at the end of the file

File: src/index.js [hunk 2]
```diff
@@ -55,7 +54,9 @@ app.use('/vendors', vendorRouter);
 app.use('/', roleRoute);
 app.use('/', permissionRoute);
 app.listen(port, async () => {
-  console.log('Database Connected!');
+  console.log(process.env.CLIENT_ID, process.env.FACEBOOK_APP_ID);
```
Comment (dushimeemma) [inline src/index.js]: Do we need this log?

**Review by dushimeemma — COMMENTED** [review]
I am enable to run your branch in my local environment, would you mind to check that?

File: src/services/user.js
Comment (Honore123) [inline src/services/user.js:1]: Delete this file

File: src/database/migrations/20230409220608-create-user.js~HEAD [hunk 3]
```diff
@@ -39,12 +43,29 @@ module.exports = {
...
         type: Sequelize.STRING,
+      },
+      roleId: {
+        type: Sequelize.INTEGER,
         allowNull: false,
       },
-      role: {
+      status: {
         type: Sequelize.STRING,
+<<<<<<< HEAD:src/database/migrations/20230409220608-create-user.js~HEAD
```
Comment (Honore123) [inline src/database/migrations/20230409220608-create-user.js~HEAD]: delete these conflict text

File: src/database/migrations/20230409220608-create-user.js~HEAD [hunk 4]
```diff
@@ -39,12 +43,29 @@ module.exports = {
...
         allowNull: false,
+        defaultValue: 'active',
+      },
+      confirmationCode: {
+        type: Sequelize.INTEGER,
+      },
+      confirmed: {
+        type: Sequelize.BOOLEAN,
       },
+>>>>>>> 0be3c68 (feature(created) route for product searching):src/database/migrations/20220409220608-create-user.js
```
Comment (Honore123) [inline src/database/migrations/20230409220608-create-user.js~HEAD]: Delete this also

**Review by Honore123 — DISMISSED** [review]
🚀

**Review by Charlesntwari — DISMISSED** [review]

File: src/tests/images/Blue and Red Modern Real Estate Instagram Post (1).png
Comment (dushimeemma) [inline src/tests/images/Blue and Red Modern Real Estate Instagram Post (1).png:1]: Would clarify about the reason of this hardcoded images?

Comment (Niyobuzima) [reply src/tests/images/Blue and Red Modern Real Estate Instagram Post (1).png:1]: these were from another branch it's old and not needed i just forgot to delete them

**Review by Charlesntwari — DISMISSED** [review]

**Comment by Charlesntwari** [comment]
good work @Niyobuzima

**Review by kunda4 — DISMISSED** [review]

**Review by Honore123 — DISMISSED** [review]

**Review by dushimeemma — COMMENTED** [review]
Write a clear commit message according to [this wiki](https://github.com/atlp-rwanda/engineering-playbook/wiki) @Niyobuzima

**Review by Charlesntwari — DISMISSED** [review]

**Review by Honore123 — APPROVED** [review]

**Review by kunda4 — APPROVED** [review]

**Review by Charlesntwari — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]
Good job 🥇

**Pull request merged**